pygfx/objects/_base.py

```diff
@@ -246,6 +249,15 @@ def material(self):
     def material(self, material):
         self._store.material = material
 
+    @property
+    def receive_shadow(self):
+        """Whether this object receive shadows. Default False."""
+        return self._store.receive_shadow
+
+    @receive_shadow.setter
```

Comment:
Changing the value of `receive_shadow` will cause the shader of the wobject to be recompiled, but `cast_shadow` will not. `cast_shadow` only determines whether the wobject participates in rendering shadow maps.

But, `cast_shadow` of `Light` objects is an exception, which will affect shader logic, so for `Light` objects, `cast_shadow` is passed to the shader as uniform.

Reply:
> `cast_shadow` only determines whether the wobject participates in rendering shadow maps.

Then this would affect the environment right? So we still want it to be trackable?

Reply:
> Then this would affect the environment right? So we still want it to be trackable?

No, it isn't.

The shadow map is rendered every frame. When rendering, the objects in the scene which cast shadows are rendered one by one.  Other objects (which not cast shadows) will be skipped.

Just like rendering the scene itself, When rendering,  only the objects that need to be rendered in the scene are rendered one by one, and the invisible objects are directly skipped.